### PR TITLE
Replace SuSEfirewall by firewalld in proxy tests

### PR DESCRIPTION
**Pull request by aaannz**
## What does this PR change?

Replace deprecated SuSEfirewall2 package by firewalld in proxy testsuite.

## Links

Fixes https://github.com/SUSE/spacewalk/issues/7182

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [x] Re-run test "changelog_test"  
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"

### CODE REVIEW

File: testsuite/features/step_definitions/command_steps.rb [hunk 1]
```diff
@@ -624,10 +624,7 @@
 end
 
 When(/^I open avahi port on the proxy$/) do
-  sed = 's/FW_DEV_EXT=""/FW_DEV_EXT="eth0"/'
-  $proxy.run("sed -i '#{sed}' /etc/sysconfig/SuSEfirewall2")
-  sed = 's/FW_CONFIGURATIONS_EXT=""/FW_CONFIGURATIONS_EXT="avahi"/'
-  $proxy.run("sed -i '#{sed}' /etc/sysconfig/SuSEfirewall2")
+  $proxy.run('firewall-cmd --permanent --add-service=mdns')
```
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb]: TBH I'm not sure if firewalld will manage those options
- FW_DEV_EXT="eth0" 
- FW_CONFIGURATIONS_EXT="avahi"
by itself, or we need to add some configuration for it.

Comment (aaannz) [reply testsuite/features/step_definitions/command_steps.rb]: mdns service is under which avahi is known in firewalld world.
See:
/etc/sysconfig/SuSEfirewall2.d/services/avahi
```ini
## Name: Zeroconf/Bonjour Multicast DNS
## Description: Zeroconf/Bonjour Multicast DNS (mDNS) ports for Service Discovery (DNS-SD)

# Allow broadcasting on UDP port 5353.
# mdnsresponder (5354) port is not used by Avahi
UDP="mdns"
BROADCAST="mdns"
```
vs

/usr/lib/firewalld/services/mdns.xml
```xml
<?xml version="1.0" encoding="utf-8"?>
<service>
  <short>Multicast DNS (mDNS)</short>
  <description>mDNS provides the ability to use DNS programming interfaces, packet formats and operating semantics in a small network without a conventional DNS server. If you plan to use Avahi, do not disable this option.</description>
  <port protocol="udp" port="5353"/>
  <destination ipv4="224.0.0.251" ipv6="ff02::fb"/>
</service>
```

I'm not 100% sure about the eth0. In my test setup adding this to the default zone was sufficient. I'll do another test run if it is really needed or not.

**Comment by aaannz** [comment]
@srbarrios I updated PR with what passes core tests suite. Please, take a look. Also do you tests proxy reboot and if it works after that?

File: testsuite/features/step_definitions/command_steps.rb (hunk 1 above)
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb]: > @srbarrios I updated PR with what passes core tests suite. Please, take a look. Also do you tests proxy reboot and if it works after that?



> @srbarrios I updated PR with what passes core tests suite. Please, take a look. Also do you tests proxy reboot and if it works after that?

If you passed the testsuite in local and it works, then LGTM, just ask your room colleague @lkotek  :)

File: testsuite/features/core_proxy_branch_network.feature [hunk 2]
```diff
@@ -113,11 +113,10 @@ Feature: Setup SUSE Manager for Retail branch network
...
 
 @proxy
-@private_net
-  Scenario: Enable avahi on the branch server
+ @private_net
+  Scenario: Enable repositories for installing branch services
     When I enable repositories before installing branch server
-    And I install package "SuSEfirewall2 expect" on this "proxy"
-    And I open avahi port on the proxy
+    Then I install package "expect" on this "proxy"
```
Comment (Bischoff) [inline testsuite/features/core_proxy_branch_network.feature]: `And` please

File: testsuite/features/core_proxy_branch_network.feature [hunk 3]
```diff
@@ -113,11 +113,10 @@ Feature: Setup SUSE Manager for Retail branch network
     Then I should see a "Formula saved!" text
 
 @proxy
-@private_net
-  Scenario: Enable avahi on the branch server
+ @private_net
```
Comment (Bischoff) [inline testsuite/features/core_proxy_branch_network.feature]: No space please

File: testsuite/features/core_proxy_branch_network.feature [hunk 4]
```diff
@@ -113,11 +113,10 @@ Feature: Setup SUSE Manager for Retail branch network
...
+  Scenario: Enable repositories for installing branch services
```
Comment (Bischoff) [inline testsuite/features/core_proxy_branch_network.feature]: This is not what this is doing anymore.

I suggest:
```cucumber
Scenario: Install expect package that is needed for testing Retail
```

Comment (aaannz) [reply testsuite/features/core_proxy_branch_network.feature]: I don't know why expect is needed, but I need repositories enabled for next step when highstate is applied.

Comment (Bischoff) [reply testsuite/features/core_proxy_branch_network.feature]: Ah, so we need both. I thought we needed the repositories to install SUSEfirewall2 and expect.

OK, forget about expect and keep that title (or put it in a separate scenario).

**Comment by mcalmer** [comment]
Not sure why changelog_test still fail

**Pull request merged**